### ISSUE TITLE
Multiple errors on initial test run (newbie needs help)

### ISSUE THREAD

**Issue by dialex** [stryker-mutator/stryker#144]
I'm having this [same issue](https://github.com/stryker-mutator/stryker/issues/142) but I think the cause may be different.

I tried running `npm run stryker` but I got this:

```
env: node\r: No such file or directory
npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/Cellar/node/6.4.0/bin/node" "/usr/local/bin/npm" "run" "stryker"
npm ERR! node v6.4.0
npm ERR! npm  v3.10.3
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
```

So I tried `node node_modules/stryker/bin/stryker -c stryker.conf.js` and still no success:

```
[2016-08-24 11:28:02.855] [WARN] TestSelectorOrchestrator - Could not find test selector `mocha` (based on the configured testFramework). Stryker will continue without the ability to select individual tests, thus running all test for every generated mutant. Set `testSelector` option explicitly to `null` to ignore this warning. Did you forget to load a plugin? Known test selectors: ["jasmine"].
[2016-08-24 11:28:02.867] [WARN] InputFileResolver - Globbing expression "test/**/*.js" did not result in any files.
[2016-08-24 11:28:02.900] [INFO] InputFileResolver - Found 102 file(s) to be mutated.
[2016-08-24 11:28:03.133] [ERROR] Stryker - One or more tests errored in the initial test run:
        [object Object]
[2016-08-24 11:28:03.133] [ERROR] stryker-cli - an error occurred Error: There were failed tests in the initial test run
    at /Users/PATH_TO_CURRENT_FOLDER/node_modules/stryker/src/Stryker.js:55:23
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

**Any idea what I'm doing wrong here?** Help is much appreciated, as I'm trying mutation test for the first time and I wanted to show this Stryker tool to my team. Thanks in advance.

---

Files:

``` json
// package.json
{
  "scripts": {
    (...)
    "stryker": "stryker -c stryker.conf.js"
  },
  "devDependencies": {
    (...)
    "chai": "^3.4.1",
    "mocha": "^2.4.5",
    "sinon": "^1.17.3",
    "stryker": "^0.4.2",
    "stryker-api": "^0.2.0",
    "stryker-html-reporter": "^0.2.0",
    "stryker-mocha-runner": "^0.1.0",
}
```

``` js
// stryker.conf.js
module.exports = function (config) {
  config.set({
    files: [
      // Add your files here, this is just an example:
      {pattern: 'src/test/**/*.js', mutated: true, included: false},
      'test/**/*.js'
    ],
    testRunner: 'mocha',
    testFramework: 'mocha',
    reporter: ['html', 'progress']
  });
};

```

**Comment by nicojs** [comment]
@dialex thanks for your issue!

Yes, it seems that you are suffering from the same issue as #142 . However, you should still be able to see the errors. Did you try to run stryker with `logLevel` set to `trace`? Changing the logLevel is explained here: https://github.com/stryker-mutator/stryker#log-level. Please try to post the underlying cause for the error here so we can help you further.

Sorry for the late response, I will be more active now that i'm back after a vacation.

**Comment by dialex** [comment]
No success with `npm run stryker` so I used again the `node` command. The output is below, I tried to separate it into logical blocks. I can't see any relevant info though 😢

```
$ node node_modules/stryker/bin/stryker --logLevel trace -c stryker.conf.js

[2016-09-16 17:20:15.657] [DEBUG] ConfigReader - Loading config stryker.conf.js
[2016-09-16 17:20:15.660] [DEBUG] PluginLoader - Loading stryker-* from /Users/dnunes/IdeaProjects/XXXXXX/node_modules
[2016-09-16 17:20:15.663] [DEBUG] PluginLoader - Loading plugins stryker-html-reporter (matched with expression stryker-*)
[2016-09-16 17:20:15.663] [DEBUG] PluginLoader - Loading plugins stryker-mocha-runner (matched with expression stryker-*)
[2016-09-16 17:20:15.663] [DEBUG] PluginLoader - Loading plugins stryker-html-reporter
[2016-09-16 17:20:15.742] [DEBUG] PluginLoader - Loading plugins stryker-mocha-runner
[2016-09-16 17:20:15.769] [DEBUG] Stryker - Using config: {"logLevel":"trace","testFramework":"mocha","testRunner":"mocha","timeoutMs":5000,"timeoutFactor":1.5,"plugins":["stryker-*"],"port":9234,"reporter":["html","progress"],"files":[{"pattern":"src/test/**/*.js","mutated":true,"included":false},"test/**/*.js"],"configFile":"stryker.conf.js"}

[2016-09-16 17:20:15.770] [WARN] TestSelectorOrchestrator - Could not find test selector `mocha` (based on the configured testFramework). Stryker will continue without the ability to select individual tests, thus running all test for every generated mutant. Set `testSelector` option explicitly to `null` to ignore this warning. Did you forget to load a plugin? Known test selectors: ["jasmine"].
[2016-09-16 17:20:15.779] [WARN] InputFileResolver - Globbing expression "test/**/*.js" did not result in any files.

[2016-09-16 17:20:15.803] [INFO] InputFileResolver - Found 108 file(s) to be mutated.
[2016-09-16 17:20:15.803] [DEBUG] TestRunnerOrchestrator - Creating test runner 0 using settings {port: 9234, coverageEnabled: true}

[2016-09-16 17:20:15.929] [TRACE] IsolatedTestRunnerAdapter - [2016-09-16 17:20:15.925] [DEBUG] PluginLoader - Loading stryker-* from /Users/dnunes/IdeaProjects/XXXXXX/node_modules
[2016-09-16 17:20:15.931] [TRACE] IsolatedTestRunnerAdapter - [2016-09-16 17:20:15.931] [DEBUG] PluginLoader - Loading plugins stryker-html-reporter (matched with expression stryker-*)
[2016-09-16 17:20:15.931] [TRACE] IsolatedTestRunnerAdapter - [2016-09-16 17:20:15.931] [DEBUG] PluginLoader - Loading plugins stryker-mocha-runner (matched with expression stryker-*)
[2016-09-16 17:20:15.931] [TRACE] IsolatedTestRunnerAdapter - [2016-09-16 17:20:15.931] [DEBUG] PluginLoader - Loading plugins stryker-html-reporter
[2016-09-16 17:20:16.012] [TRACE] IsolatedTestRunnerAdapter - [2016-09-16 17:20:16.012] [DEBUG] PluginLoader - Loading plugins stryker-mocha-runner
[2016-09-16 17:20:16.046] [TRACE] IsolatedTestRunnerAdapter - [2016-09-16 17:20:16.046] [DEBUG] StrykerMochaReporter - Starting Mocha test run
[2016-09-16 17:20:16.046] [TRACE] IsolatedTestRunnerAdapter - [2016-09-16 17:20:16.046] [DEBUG] StrykerMochaReporter - Mocha test run completed: 0/0 passed

[2016-09-16 17:20:16.049] [ERROR] Stryker - One or more tests errored in the initial test run:
	[object Object]
[2016-09-16 17:20:16.049] [ERROR] stryker-cli - an error occurred Error: There were failed tests in the initial test run
    at /Users/dnunes/IdeaProjects/XXXXXX/node_modules/stryker/src/Stryker.js:55:23
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

Now it's my time to apologise for the delay!

**Comment by nicojs** [comment]
Sorry to keep you waiting.

I think mocha is reporting test failures, because it discovered zero tests. As you can see `Globbing expression "test/**/*.js" did not result in any files.`. Are you sure the test folder is not empty? And that you are starting the test from the correct directory. The expression `test/**/*.js` seems correct to me.

**Comment by dialex** [comment]
Nice, my bad, I changed the configuration for

`{pattern: 'src/test/**/*.js', mutated: true, included: false}, 'src/test/**/*.js'`

so now I think I got the config right. (check this image to see the [folder structure](https://cloud.githubusercontent.com/assets/2140450/19395963/5fb7a1bc-9238-11e6-9b05-d27e498c1505.png))

However, doing that only muted the warning you mentioned. Is it because it doesn't recognise my js files as tests? I'm using Mocha, Enzyme, Chai, React, Redux. A typical test looks like this:

```
describe('Module desc, ', () => {

  it("test desc", () => {
    // Given
    const store = prepareStore();

    // When
    const container = mount(<SomeElement/>);

    // Then
    something.text().should.equal("text goes here");
  });
```

Don't you find that `Could not find test selector 'mocha'` suspicious?

**Comment by simondel** [comment]
A test selector for Stryker is a piece of code that is capable of ensuring that only certain tests are ran.

For example: a code coverage result shows that only test 3, 8 and 42 cover mutant A. If a test selector is present, Stryker will tell it that only those tests have to be executed. If no test selector is present,  all of your tests will be executed for every mutant.

We don't have a test selector for Mocha right now, because the stryker-mocha-runner is not capable of generating code coverage.

**Comment by dialex** [comment]
Thanks for the insight Simon! So that means that even without Mocha, Stryker should be able to execute all tests for all mutants. That would just take more time but should work. Right now no test/mutator is running at all 😕

**Comment by nicojs** [comment]
@dialex since the 0.5 release of Stryker (with stryker-api 0.4) we changed the api. You now don't use a `testSelector`, instead you can select the `coverageAnalysis` explicitly. We also updated the error messages and it is now possible to exclude files (using a `!` in the files array, see [the stryker readme](https://github.com/stryker-mutator/stryker)).

Sorry we didn't respond earlier. We will try to do better from now on.

**Comment by dialex** [comment]
### The version

Well I used `npm install stryker stryker-api --save-dev` to update my version of stryker, and now my `package.json` looks like

```json
"stryker": "^0.4.5",
"stryker-api": "^0.2.1",
```

...shouldn't it be using v0.5.4 (currently the latest on github)?

------

### The good

Finally `npm run stryker` is returning the same as `node node_modules/stryker/bin/stryker -c stryker.conf.js`

------

### The bad

I added to the `stycker.conf.js` file these two lines to match your README example.

```js
coverageAnalysis: 'perTest',
plugins: ['stryker-mocha-runner', 'stryker-html-reporter']
```

This is part of my current `package.json` file

```json
"test": "mocha --compilers js:babel-core/register --require ./src/test/test_helper.js 'src/test/**/*.@(js)' --recursive",
"test:watch": "mocha --compilers js:babel-core/register --require ./src/test/test_helper.js 'src/test/**/*.@(js)' --recursive --watch",
"stryker": "stryker -c stryker.conf.js",
```

Then I ran `npm run stryker` and got this, which is basically the same issue I always had.

```
[WARN] TestSelectorOrchestrator - Could not find test selector `mocha` (based on the configured testFramework). Stryker will continue without the ability to select individual tests, thus running all test for every generated mutant. Set `testSelector` option explicitly to `null` to ignore this warning. Did you forget to load a plugin? Known test selectors: ["jasmine"].
[INFO] InputFileResolver - Found 135 file(s) to be mutated.
[INFO] TestRunnerOrchestrator - Starting initial test run. This may take a while.
[ERROR] Stryker - One or more tests errored in the initial test run:
  [object Object]
[ERROR] stryker-cli - an error occurred Error: There were failed tests in the initial test run
    at /Users/dnunes/IdeaProjects/Vocalink/app/ops-console-client/node_modules/stryker/src/Stryker.js:58:23
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

What now?

**Comment by simondel** [comment]
Strange.. I just tested it and I get the latest versions of Stryker.

Could you change your package.json to:
```
....
"stryker": "^0.5.4",
"stryker-api": "^0.4.1",
....
```

Then remove your node_modules folder and do an `npm install`?

**Comment by dialex** [comment]
I did the update you requested. Then I had to update `package.json` to the new syntax:

```json
"stryker": "stryker run --coverageAnalysis all stryker.conf.js"
```

Then I got an error saying: _Configured coverage analysis "perTest" requires there to be a testFramework configured. Either configure a testFramework or set coverageAnalysis to "all" or "off"._ So I did.

Now I get 143 error messages, for the 143 test files:
`[ERROR] CoverageInstrumenterStream - Error while instrumenting file "/.../src/test/.../SomeTestFile.js", error was: Error: Line 1: Unexpected token
Failed to parse file`.
And then
`[ERROR] Stryker - One or more tests errored in the initial test run: TypeError: Cannot read property 'Error' of undefined`

I can't see anything wrong with those files. Mocha executes them just fine.

**Comment by simondel** [comment]
Could you post what's in your stryker.conf.js right now? We had some breaking changes with the `0.5.0` release of Stryker.

What's on line 1 of `"/.../src/test/.../SomeTestFile.js"`?

**Comment by dialex** [comment]
This is my current `stryker.conf.js`:

```js
module.exports = function (config) {
  config.set({
    files: [
      // Add your files here, this is just an example:
      {pattern: 'src/test/**/*.js', mutated: true, included: false},
      'src/test/**/*.js'
    ],
    testRunner: 'mocha',
    testFramework: 'mocha',
    reporter: ['html', 'progress'],
    coverageAnalysis: 'perTest',
    plugins: ['stryker-mocha-runner', 'stryker-html-reporter']
  });
};
```

And this is an example of line 1 of my `*.js` tests:
`import React from "react";`
or
`import {expect, assert} from "chai";`

**Comment by simondel** [comment]
You could set `coverageAnalysis` to `'off'` which should fix this for you. We don't properly support mutating files that include import statements yet. We'll fix/discuss that in #258
